domain/status/status.go

```diff
@@ -87,6 +87,40 @@ func DecodeK8sPodStatus(s int) (K8sPodStatusType, error) {
 	}
 }
 
+// ModelStatus represents the health status of a Juju model.
+type ModelStatusType string
+
+const (
+	ModelStatusSuspended  ModelStatusType = "suspended"
+	ModelStatusAvailable  ModelStatusType = "available"
+	ModelStatusDestroying ModelStatusType = "destroying"
+	ModelStatusBusy       ModelStatusType = "busy"
+)
+
+// ModelStatus represents the current status of a model.
+type ModelStatus struct {
+	// Status is the current status of the model.
+	Status ModelStatusType
+	// Message is a human-readable message that describes the current status of the model.
+	Message string
+	// Reason is a human-readable message that describes the reason for the current status of the model.
+	Reason string
+	// Since is the time when the model entered the current status.
+	Since time.Time
+}
+
+// ModelStatusContext describes the context used to determine the status of a model.
+type ModelStatusContext struct {
+	// Destroying indicates if the model is in the process of being destroyed.
+	Destroying bool
+	// Migrating indicates if the model is in the process of being migrated.
+	Migrating bool
+	// InvalidCloudCredential indicates if the model's cloud credential is invalid.
+	InvalidCloudCredential bool
```

Comment:
HasInvalidCloudCredential is a better name.

Reply:
Hi Kelvin, I decided to drop the prefix to maintain consistency based on the comment made by Simon. Perhaps let me add a prefix for the other two instead to make this clearer.

https://github.com/juju/juju/pull/19895#discussion_r2125920138